modules/iam/data.tf

```diff
@@ -2,7 +2,444 @@ data "aws_caller_identity" "current" {}
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+  }
+
+  statement {
+    sid       = "PutCloudWatchMetrics"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["cloudwatch:PutMetricData"]
+  }
+
+  statement {
+    sid    = "SNS"
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:sns:*:*:${var.project}-${var.environment}",
+      "arn:aws:sns:*:*:${var.project}-dlq-${var.environment}",
+    ]
+
+    actions = [
+      "sns:Publish",
+      "sns:Subscribe",
+    ]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+    ]
+  }
+
+  statement {
+    sid       = "GetAccountName"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["iam:ListAccountAliases"]
+  }
+
+  statement {
+    sid       = "DescribeRegions"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["ec2:DescribeRegions"]
+  }
+
+  statement {
+    sid       = "CloudFormation"
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "cloudformation:DescribeStacks",
+      "cloudformation:ListStacks",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "accounts" {
+  statement {
+    sid       = "WriteToCloudWatchLogs"
+    effect    = "Allow"
+    resources = ["arn:aws:logs:*:*:*"]
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+  }
+
+  statement {
+    sid       = "AssumeSecurityAuditRole"
+    effect    = "Allow"
+    resources = ["arn:aws:iam::*:role/${var.security_audit_role_name}"]
+    actions   = ["sts:AssumeRole"]
+  }
+
+  statement {
+    sid       = "PutCloudWatchMetrics"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["cloudwatch:PutMetricData"]
+  }
+
+  statement {
+    sid    = "SNS"
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:sns:*:*:${var.project}-${var.environment}",
+      "arn:aws:sns:*:*:${var.project}-dlq-${var.environment}",
+    ]
+
+    actions = [
+      "sns:Publish",
+      "sns:Subscribe",
+    ]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+    ]
+  }
+
+  statement {
+    sid       = "DynamoDB"
+    effect    = "Allow"
+    resources = ["${local.ddb_table_arn}"]
+
+    actions = [
+      "dynamodb:PutItem",
+      "dynamodb:Query",
+      "dynamodb:Scan",
+      "dynamodb:UpdateItem",
+    ]
+  }
+
+  statement {
+    sid       = "StartStateMachine"
+    effect    = "Allow"
+    resources = ["${var.state_machine_arn}"]
+    actions   = ["states:StartExecution"]
+  }
+}
+
+data "aws_iam_policy_document" "basic" {
+  statement {
+    sid       = "WriteToCloudWatchLogs"
+    effect    = "Allow"
+    resources = ["arn:aws:logs:*:*:*"]
```

Comment:
I know this was here earlier, however, does this block require permissions to ALL logs instead of specific log groups ?

Reply:
Added the only groups that I could find here.